src/newdb.cpp

```diff
@@ -740,27 +745,57 @@ bool load_char(const char *name, char_data *ch, bool logon)
 
         auto_repair_obj(obj, buf3);
 
-        if (inside > 0) {
-          if (inside == last_in)
-            last_obj = last_obj->in_obj;
-          else if (inside < last_in)
-            while (inside <= last_in && last_obj) {
-              last_obj = last_obj->in_obj;
-              last_in--;
+        // Since we're now reading rows from the db in reverse order, in order to fix the stupid reordering on
+        // every binary execution, the previous algorithm did not work, as it relied on getting the container obj
+        // first and place subsequent objects in it. Since we're now getting it last, and more importantly we get
+        // objects at deeper nesting level in between our list, we save all pointers to objects along with their
+        // nesting level in a vector container and once we found the next container (inside < last_inside) we
+        // move the objects with higher nesting level to container and proceed. This works for any nesting depth.
+        if (inside > 0 || (inside == 0 && inside < last_inside)) {
+          //Found our container?
+          if (inside < last_inside) {
+            if (inside == 0)
+              obj_to_cyberware(obj, ch);
+              
+            auto it = std::find_if(contained_obj.begin(), contained_obj.end(), find_level(inside+1));
+            while (it != contained_obj.end()) {
+              obj_to_obj(it->obj, obj);
+              contained_obj.erase(it);
             }
-          if (last_obj)
-            obj_to_obj(obj, last_obj);
-        } else obj_to_cyberware(obj, ch);
-        last_in = inside;
-        last_obj = obj;
+              
+            if (inside > 0) {
+              contained_obj_entry.level = inside;
+              contained_obj_entry.obj = obj;
+              contained_obj.push_back(contained_obj_entry);
+            }
+          }
+          else {
+            contained_obj_entry.level = inside;
+            contained_obj_entry.obj = obj;
+            contained_obj.push_back(contained_obj_entry);
+          }
+          last_inside = inside;
+        } else
+          obj_to_cyberware(obj, ch);
+          
+        last_inside = inside;
       }
     }
+    //Failsafe. If something went wrong and we still have objects stored in the vector, dump them in inventory.
+    if (!contained_obj.empty()) {
+      for (auto it : contained_obj)
+        obj_to_char(it.obj, ch);
+      
+      contained_obj.clear();
+      snprintf(buf2, sizeof(buf2), "Load error:  Objects in ware containers found with invalid containers for Char ID: %ld. Dumped in inventory.", GET_IDNUM(ch));
+      mudlog(buf2, NULL, LOG_SYSLOG, TRUE);
+    }
     mysql_free_result(res);
   }
   {
     struct obj_data *obj;
     int vnum = 0;
-    snprintf(buf, sizeof(buf), "SELECT * FROM pfiles_bioware WHERE idnum=%ld;", GET_IDNUM(ch));
+    snprintf(buf, sizeof(buf), "SELECT * FROM pfiles_bioware WHERE idnum=%ld ORDER BY Vnum;", GET_IDNUM(ch));
```

Comment:
Why order by vnum?

Reply:
Temporary trick to make bioware output consistent, since there's no `posi` field there to order by.